### PR TITLE
Add UI access tests

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Add E2E tests to ensure the UI can be accessed properly (tested in Kubernetes and OpenShift)

## Short description of the changes
- Add an E2E test to check the UI can be accessed properly in `allinone` and `production` deployment strategies
- Refactor the code to get a token to access programmatically Jaeger query
  - Create `get-token.sh` to get in a easier way the token
  - Refactor multiple templates to make it easier to centralize the logic to get the token
- Create the `ui` E2E test suite
- Remove the `allinone` E2E test suite. It contained 3 tests:
  - `ingress`: it checks are now done by the `ui/allinone` and `ui/production` E2E tests
  - `uiconfig`: it checks are now done by the `ui/allinone` and `ui/production` E2E tests
  - `smoke`: it was tested before by the the `examples/simplest` E2E test
- The `uiconfig` executable is not part anymore of the `asserts` Docker image: it was used just by 2 tests and it takes a lot of time to build. The build of the `asserts` Docker image should be faster now

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -167,7 +167,7 @@ cert-manager: cmctl
 	cmctl check api --wait=5m
 
 undeploy-cert-manager:
-	kubectl delete -f https://github.com/jetstack/cert-manager/releases/download/v${CERTMANAGER_VERSION}/cert-manager.yaml
+	kubectl delete --ignore-not-found=true -f https://github.com/jetstack/cert-manager/releases/download/v${CERTMANAGER_VERSION}/cert-manager.yaml
```
Comment (frzifus) [inline Makefile:170]:
```suggestion
	kubectl delete --ignore-not-found=$(ignore-not-found) -f https://github.com/jetstack/cert-manager/releases/download/v${CERTMANAGER_VERSION}/cert-manager.yaml
```

File: tests/cmd-utils/uiconfig/main.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	if expectedContent == "" {
+		logrus.Fatalln("EXPECTED_CONTENT env variable could not be empty")
+	}
+
+	logrus.Info("Querying ", hostName, "...")
+
+	found := false
+
+	err := utils.TestGetHTTP(hostName, params, func(_ *http.Response, body []byte) (done bool, err error) {
+		if strings.Contains(string(body), expectedContent) {
```
Comment (frzifus) [inline tests/cmd-utils/uiconfig/main.go:39]:
```suggestion
		found := strings.Contains(string(body), expectedContent)
```

**Review by frzifus — COMMENTED** [review]
LGTM, only a few minor things - up to @rubenvp8510

File: Makefile (hunk 1 above)
Comment (iblancasa) [reply Makefile:170]: Why is this needed?

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**